src/ripple/app/ledger/impl/LedgerDeltaAcquire.cpp

```diff
@@ -174,38 +173,37 @@ LedgerDeltaAcquire::processData(
             complete_ = true;
             orderedTxns_ = std::move(orderedTxns);
             JLOG(journal_.debug()) << "ready to replay " << hash_;
-            notifyTasks(sl);
+            notify(sl);
             return;
         }
     }
 
     failed_ = true;
     JLOG(journal_.error())
         << "failed to create a (info only) ledger from verified data " << hash_;
-    notifyTasks(sl);
+    notify(sl);
 }
 
 void
-LedgerDeltaAcquire::addTask(std::shared_ptr<LedgerReplayTask>& task)
+LedgerDeltaAcquire::addDataCallback(
+    InboundLedger::Reason reason,
+    OnDeltaDataCB&& cb)
 {
     ScopedLockType sl(mtx_);
-    tasks_.emplace_back(task);
-
-    auto reason = task->getTaskParameter().reason;
+    dataReadyCallbacks_.emplace_back(cb);
```

Comment:
Do you want to move from `cb`? If so, I think you want `dataReadyCallbacks_.emplace_back(std::move(cb));`

Reply:
Thanks, changed.